tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -544,6 +544,31 @@ void TRTEngineOp::ComputeAsync(OpKernelContext* ctx,
   }
 }
 
+// Get the binding index of a tensor in an engine.
+// The binding index is looked up using the tensor's name and the profile idx.
+// Profile idx should be set to zero, if we do not have optimization profiles.
+Status GetTrtBindingIndex(const char* tensor_name, int profile_idx,
+                          const nvinfer1::ICudaEngine* cuda_engine,
+                          int* binding_idx) {
+  *binding_idx = cuda_engine->getBindingIndex(tensor_name);
+  if (*binding_idx == -1) {
+    const string msg = StrCat("Input node ", tensor_name, " not found");
+    LOG(ERROR) << msg;
+    return errors::NotFound(msg);
+  }
+#if IS_TRT_VERSION_GE(6, 0, 0, 0)
+  int n_profiles = cuda_engine->getNbOptimizationProfiles();
+#else
+  int n_profiles = 1;
+#endif
+  // If we have more then one optimization profiles then the binding idx
+  // depends on the profile number
```

Comment:
Can we replace this comment with this:
binding_index_within_engine = binding_index_within_profile + profile_index * bindings_per_profile

Reply:
Added this comment.